flax/nnx/nn/attention.py

```diff
@@ -330,6 +334,7 @@ def __init__(
     qkv_dot_general_cls: Any = None,
     out_dot_general_cls: Any = None,
     rngs: rnglib.Rngs,
+    keep_rngs: bool = True,
```

Comment:
maybe we make this `False` by default?

Reply:
It's true that in both cases it is BC-breaking. It would be good to make users aware of the change with a warning?
Or we can have a three-state flag: None, True, False, with None doing the previous non-breaking behavior and setting default value to None + warning saying about a future change to `keep_rngs=False` as default. What do you think?